### PR TITLE
rocsparse: init at 2.3.2-5.3.1

### PR DESCRIPTION
**Pull request by Madouura**
###### Description of changes
Tracking: #197885
Third part in a whole bunch of porting...
``rocsparse.passthru.tests`` will likely always fail due to it being in a vm, but I have confirmed ~~a subset of tests (there are like 284K of them...) generated do work on hardware~~ all tests work!.
We may want to consider how many of these tests we want to do whenever it is possible to use GPU acceleration (emulated or real) in testing vms.
Depends on #197337

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Madouura** [comment]
Cleaned up some things.

**Comment by Madouura** [comment]
Changed some overly-verbose paths.

**Comment by Madouura** [comment]
Before not all tests were passing after I ran it longer because ``${rocsparse.test}`` was missing the generated matrices directory. That is fixed now.

```console
[----------] Global test environment tear-down
[==========] 282847 tests from 328 test suites ran. (9524521 ms total)
[  PASSED  ] 282847 tests.
```

File: pkgs/development/libraries/rocsparse/default.nix [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+, hip
+, gfortran
+, git
+, fetchzip ? null
+, gtest ? null
+, boost ? null
+, python3Packages ? null
+, buildTests ? false
+, buildBenchmarks ? false # Seems to depend on tests
+, rocmVersion ? "5.3.1" # rocSPARSE depends on specific ROCm versions per-release
```
Comment (lovesegfault) [inline pkgs/development/libraries/rocsparse/default.nix:20]: Same as on the other PRs, use `hip.version` directly.

**Comment by Madouura** [comment]
Rolled into #198770

**Pull request closed without merging**